32.JS_Functions/index.js

```diff
@@ -0,0 +1,104 @@
+// ----- I. Minimun
+
+// 1. Напиши всі можливі варіанти створення функцій.
+// 2. Створи функцію, яка буде виводити кількість переданих їй аргументів.
+// 3. Напиши функцію, яка приймає 2 числа і повертає :
+//      -1, якщо перше число менше, ніж друге;
+//      1 - якщо перше число більше, ніж друге;
+//      0 - якщо числа рівні.
+// 4. Напиши функцію, яка обчислює факторіал переданого їй числа.
+// 5. Напиши функцію, яка приймає три окремі цифри і перетворює їх в одне число. Наприклад: цифри 1, 4, 9 перетворяться в число 149.
+// 6. Напиши функцію, яка приймає довжину і ширину прямокутника і обчислює його площу. Якщо в функцію передали 1 параметр, то вона обчислює площу квадрата.
+
+// ---------------------------------------------------------------------
+
+// ----- II. Normal
+
+// 1. Напиши функцію, яка перевіряє, чи є передане їй число “досконалим числом”. Досконале число - це число, яке дорівнює сумі всіх своїх дільників.
+// 2. Напиши функцію, яка приймає мінімальне і максимальне значення для діапазону, і виводить тільки ті числа з діапазону, які є досконалими. Використовуй написану раніше функцію, щоб дізнатися, чи є це число досконалим.
+
+// --------- ANSWERS ---------
+
+// ----- I. Minimun
+
+// 1.
+// 1) Function expression = function myFunf() {};
+// 2) Function declaration = const myFunc = function() {} || const myFunc = () => {};
+// 3) IIFE (Immediately invoked function expression) = (function() {
+// ******
+// })();
+
+// 2.
+// const myFunc = (...args) => {
```

Comment:
try also to do this with Function declaration (function) but without using ...args